core/src/main/java/org/bitcoinj/wallet/Wallet.java

```diff
@@ -389,15 +390,39 @@ public static Wallet fromWatchingKey(NetworkParameters params, DeterministicKey
         return new Wallet(params, KeyChainGroup.builder(params).addChain(chain).build());
     }
 
+     /**
+     * Creates a wallet that tracks payments to and from the HD key hierarchy rooted by the given watching key. The
+     * account path is specified.
+     * @param params The network
+     * @param watchKeyB58 The key in base58 notation
+     * @param creationTime creation time of the key (if unknown use {@link #fromWatchingKeyB58(NetworkParameters, String)}
+     * @return a new wallet
+     */
+    public static Wallet fromWatchingKeyB58(NetworkParameters params, String watchKeyB58, Instant creationTime) {
```

Comment:
Since we've deprecated the old `fromWatchingKeyB58` method, we should probably change the 1st parameter from `NetworkParameters` to `Network` as well.